### PR TITLE
add readme in control_performance_analysis

### PR DESCRIPTION
**Pull request by k0suke-murakami**
Signed-off-by: kosuke murakami <kosuke.murakami@tier4.jp>

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: control/control_performance_analysis/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## Input / Output
+
+### Input topics
+
+| Name                                               | Type                                                     | Description                                         |
+| -------------------------------------------------- | -------------------------------------------------------- | --------------------------------------------------- |
+| `/planning/scenario_planning/trajectory`           | autoware_auto_planning_msgs::msg::Trajectory             | Output trajectory from planning module.             |
+| `/control/trajectory_follower/lateral/control_cmd` | autoware_auto_control_msgs::msg::AckermannLateralCommand | Output lateral control command from control module. |
+| `/vehicle/status/steering_status`                  | autoware_auto_vehicle_msgs::msg::SteeringReport          | Steering information from vehicle.                  |
+| `/localization/odometry`                           | nav_msgs::msg::Odometry                                  | Use twist from odometry                             |
```
Comment (kmiya) [inline control/control_performance_analysis/README.md:21]:
```suggestion
| `/localization/odometry`                           | nav_msgs::msg::Odometry                                  | Use twist from odometry.                            |
```
Nit

File: control/control_performance_analysis/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+| `/tf`                                              | `tf2_msgs::msg::TFMessage                                | Extract ego pose from tf.                           |
+
+### Output topics
+
+| Name                                    | Type                                            | Description                            |
+| --------------------------------------- | ----------------------------------------------- | -------------------------------------- |
+| `/control_performance/performance_vars` | control_performance_analysis::msg::ErrorStamped | The result of the performance analysis |
```
Comment (kmiya) [inline control/control_performance_analysis/README.md:28]:
```suggestion
| Name                                    | Type                                            | Description                             |
| --------------------------------------- | ----------------------------------------------- | --------------------------------------- |
| `/control_performance/performance_vars` | control_performance_analysis::msg::ErrorStamped | The result of the performance analysis. |
```
Nit

**Review by kmiya — COMMENTED** [review]
Nits: Add periods.

File: control/control_performance_analysis/README.md (hunk 2 above)
Comment (k0suke-murakami) [reply control/control_performance_analysis/README.md]: Thank you, add period. https://github.com/tier4/AutowareArchitectureProposal.iv/pull/716/commits/d87dc706fe639d9dd375ea8bfe491fe3cebeba6d

**Comment by TakaHoribe** [comment]
@boyali If you have any thoughts to improve the document, please push directly on this branch. we'll merge this Monday next week.

**Comment by boyali** [comment]
> @boyali If you have any thoughts to improve the document, please push directly on this branch. we'll merge this Monday next week.

@TakaHoribe -san, do you want me to add how the errors are computed and how the vehicle is located on the path? These are not shown in the readme file

**Comment by TakaHoribe** [comment]
@boyali We are preparing a readme for all packages in .auto porting task. It doesn't have to be perfect, so the current one is enough. I just let you know if you want to improve readme before moving this package into autoware.universe, which is a public repo.

**Comment by boyali** [comment]
Yes it is enough for now, since it does not affect any package I think we
don’t need to put geometric computation details.

File: control/control_performance_analysis/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# control_performance_analysis
+
+## Purpose
+
+`control_performance_analysis` is the package to analyze the output of a control module.
```
Comment (boyali) [inline control/control_performance_analysis/README.md:5]: `control_performance_analysis` is the package to analyze the tracking performance of a control module.

**Review by boyali — APPROVED** [review]
The current package summary for me is sufficient. I thought to add how we are computing the errors and other simple control efforts and error energy but these are not necessary as they are simple geometrical computations.

File: control/control_performance_analysis/README.md (hunk 3 above)
Comment (k0suke-murakami) [reply control/control_performance_analysis/README.md:5]:
```suggestion
`control_performance_analysis` is the package to analyze the tracking performance of a control module.
```

**Pull request merged**